models/membership_test.go

```diff
@@ -577,65 +578,952 @@ var _ = Describe("Membership Model", func() {
+
+					dbMembership, err := GetMembershipByID(testDb, membership.ID)
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(dbMembership.GameID).To(Equal(membership.GameID))
+					Expect(dbMembership.PlayerID).To(Equal(players[0].ID))
+					Expect(dbMembership.RequestorID).To(Equal(players[0].ID))
+					Expect(dbMembership.ClanID).To(Equal(clan.ID))
+					Expect(dbMembership.Approved).To(Equal(false))
+					Expect(dbMembership.Denied).To(Equal(false))
+					Expect(dbMembership.DeletedAt).To(Equal(int64(0)))
+					Expect(dbMembership.Message).To(Equal("Please accept me"))
+
+					dbPlayer, err := GetPlayerByID(testDb, fixtures.GetEncryptionKey(), players[0].ID)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(dbPlayer.MembershipCount).To(Equal(0))
+
+					dbClan, err := GetClanByID(testDb, clan.ID)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(dbClan.MembershipCount).To(Equal(1))
+				})
+
+				It("If requestor is the player and clan.AllowApplication = true and autoJoin = true and previous deleted membership", func() {
+					game, clan, _, players, _, err := fixtures.GetClanWithMemberships(testDb, 1, 0, 0, 0, "", "")
+					Expect(err).NotTo(HaveOccurred())
+
+					game.CooldownBeforeApply = 0
+					_, err = testDb.Update(game)
+					Expect(err).NotTo(HaveOccurred())
+
+					clan.AllowApplication = true
+					clan.AutoJoin = true
+					_, err = testDb.Update(clan)
+					Expect(err).NotTo(HaveOccurred())
+
+					_, err = DeleteMembership(
+						testDb,
+						game,
+						clan.GameID,
+						players[0].PublicID,
+						clan.PublicID,
+						players[0].PublicID,
+					)
+
+					Expect(err).NotTo(HaveOccurred())
+
+					membership, err := CreateMembership(
+						testDb,
+						fixtures.GetEncryptionKey(),
+						game,
+						players[0].GameID,
+						"Member",
+						players[0].PublicID,
+						clan.PublicID,
+						players[0].PublicID,
+						"Please accept me",
+					)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(membership.ID).NotTo(BeEquivalentTo(0))
+
+					dbMembership, err := GetMembershipByID(testDb, membership.ID)
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(dbMembership.GameID).To(Equal(membership.GameID))
+					Expect(dbMembership.PlayerID).To(Equal(players[0].ID))
+					Expect(dbMembership.RequestorID).To(Equal(players[0].ID))
+					Expect(dbMembership.ClanID).To(Equal(clan.ID))
+					Expect(dbMembership.Approved).To(Equal(true))
+					Expect(dbMembership.Denied).To(Equal(false))
+					Expect(dbMembership.DeletedAt).To(Equal(int64(0)))
+					Expect(dbMembership.Message).To(Equal("Please accept me"))
+					Expect(dbMembership.ApprovedAt).To(BeNumerically("~", util.NowMilli(), 1000))
+
+					dbPlayer, err := GetPlayerByID(testDb, fixtures.GetEncryptionKey(), players[0].ID)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(dbPlayer.MembershipCount).To(Equal(1))
+
+					dbClan, err := GetClanByID(testDb, clan.ID)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(dbClan.MembershipCount).To(Equal(1))
+				})
```

Comment:
This test is related to the flow that was in error and is fixed in [PR#95](https://github.com/topfreegames/khan/pull/95). When a membership is created in an open clan, a player that was previously deleted should update `ApprovedAt` field.